include/private/gcconfig.h

```diff
@@ -484,6 +484,15 @@ EXTERN_C_BEGIN
 #   endif
 #   define I386
 #   define mach_type_known
+#   define WASM
```

Comment:
Defining here something else than <ARCH> or <OS> is not recommended. Even if it leads to some code duplication.